extensions/m_dehelper.c

```diff
@@ -40,26 +40,26 @@
 
 #include <string.h>
 
-static int mo_dehelper(struct Client *, struct Client *, int, const char **);
+static int m_dehelper(struct Client *, struct Client *, int, const char **);
 static int me_dehelper(struct Client *, struct Client *, int, const char **);
 
 static int do_dehelper(struct Client *source_p, struct Client *target_p);
 
 struct Message dehelper_msgtab = {
 	"DEHELPER", 0, 0, 0, MFLG_SLOW,
-	{mg_unreg, mg_not_oper, mg_not_oper, mg_ignore, {me_dehelper, 2}, {mo_dehelper, 2}}
+	{mg_unreg, {m_dehelper, 2}, mg_not_oper, mg_ignore, {me_dehelper, 2}, {m_dehelper, 2}}
 };
 
 mapi_clist_av1 dehelper_clist[] = { &dehelper_msgtab, NULL };
 DECLARE_MODULE_AV1(dehelper, NULL, NULL, dehelper_clist, NULL, NULL, "$Revision: 254 $");
 
-static int mo_dehelper(struct Client *client_p, struct Client *source_p, int parc, const char **parv)
+static int m_dehelper(struct Client *client_p, struct Client *source_p, int parc, const char **parv)
 {
 	struct Client *target_p;
 
-	if (!IsOperAdmin(source_p))
+	if (!IsOperDehelper(source_p))
```

Comment:
This might be a bit of a nitpick, but this ought to check for `IsAnyOper()` first and give `ERR_NOPRIVILEGES` if that fails. Compare the operwall code, https://github.com/freenode/ircd-seven/blob/1f1e857fe664641b67b61ead71d129b25b22b44c/modules/m_wallops.c#L64-L74

Having to do this manually for each command `+O` users are supposed to be able to access is admittedly rather ugly. Then again, so is having to check `IsOperFoo` manually for each command, so these might both be worth beautifying in the future.

Reply:
Added the check in 55de74d